src/profile-logic/symbolication.js

```diff
@@ -116,21 +123,23 @@ import type {
  * expecting a future call to applySymbolicationStep.
  *
  * V. Profile substitution: Invoked from from a thunk action. Processes the
- * symbolication result, groups frame addresses by func addresses, finds or
- * creates funcs for these funcAddresses, and creates a new thread
- * object with an updated frameTable, funcTable and stringTable, with the
+ * symbolication result, groups frame addresses by symbol addresses, finds or
+ * creates nativeSymbols for these symbolAddresses, and groups funcs by the
+ * function name. At the end, it creates a new thread object with an updated
+ * frameTable, funcTable, nativeSymbols and stringTable, with the
  * symbols substituted in the right places. This usually causes many funcs to
  * be orphaned (no frames will use them any more); these orphaned funcs remain
  * in the funcTable. It also creates the oldFuncToNewFuncMap.
  *
  * Re-symbolication only re-runs phases II through V. At the beginning of
- * re-symbolication, the frameTable and funcTable are in the state that the
- * previous symbolication left them in. If the previous symbolication merged
- * functions based on an incomplete symbol table, and re-symbolication has a
- * more detailed symbol table with finer-grained function symbols to work with,
- * then re-symbolication needs to split funcs up again. Splitting up funcs
- * means that a collection of frames which were all using the same func before
- * re-symbolication will be assigned to multiple funcs after re-symbolication.
+ * re-symbolication, the frameTable, funcTable and nativeSymbols are in the
+ * state that the previous symbolication left them in. If the previous
+ * symbolication merged functions based on an incomplete symbol table, and
+ * re-symbolication has a more detailed symbol table with finer-grained function
+ * symbols to work with, then re-symbolication needs to split funcs up again.
+ * Splitting up funcs means that a collection of frames which were all using the
+ * same func before re-symbolication will be assigned to multiple funcs after
+ * re-symbolication.
```

Comment:
Thanks for updating the comments to match the current implementation! They helped me a lot understand the code.